### ISSUE TITLE
Align vertx versions in Micronaut SQL and core

### ISSUE THREAD

**Issue by sdelamo** [micronaut-projects/micronaut-core#7648]
### Issue description

SQL  https://github.com/micronaut-projects/micronaut-sql/blob/master/gradle/libs.versions.toml#L11

Core
https://github.com/micronaut-projects/micronaut-core/blob/3.5.x/gradle/libs.versions.toml#L29

Users who want to use Hibernate Reactive may need to pin the vertx version to the one in SQL. 

see: https://github.com/micronaut-projects/micronaut-guides/pull/1002

**Comment by sdelamo** [comment]
@timyates can you check why vertx dependency is needed in core?

**Comment by timyates** [comment]
We use it to send Http2 post requests in the tests here

https://github.com/micronaut-projects/micronaut-core/blob/acac3f9705205ac8ffaa7528a64d4e963a5c5696/http-server-netty/src/test/groovy/io/micronaut/http/server/netty/http2/Http2PostTest.java#L77-L87

We *could* get rid of the version ref for `vertx` here 

https://github.com/micronaut-projects/micronaut-core/blob/a596061009e5e8dfe9b4a9d0aca0b285deca242b/gradle/libs.versions.toml#L427-L428

And add this to :http-server-netty:build.gradle

```
    testImplementation(platform(libs.boms.micronaut.sql))
```

But we would need to keep the version ref for `vertx-webclient` as this isn't exposed by the sql module

**Comment by timyates** [comment]
tldr; The vertx version in core is only used for tests, not exported, and it's fine if it's not aligned

**Comment by timyates** [comment]
Closing this as won't fix
